agent.py

```diff
@@ -261,18 +267,39 @@ def run(self, config=None):
             pipe_path = get_jmx_pipe_path()
             if Platform.is_windows():
                 pipe_name = pipe_path.format(pipename=SD_PIPE_NAME)
+                exists = win32pipe.WaitNamedPipe(pipe_name,
+                                                 win32pipe.NMPWAIT_USE_DEFAULT_WAIT)
+                if not exists:
+                    pipe = win32pipe.CreateNamedPipe(pipe_name, win32pipe.PIPE_ACCESS_DUPLEX,
+                                            win32pipe.PIPE_TYPE_MESSAGE | win32pipe.PIPE_WAIT,
```

Comment:
not convinced this should be `PIPE_WAIT`... we're only writing from our end, so this should be fine either way..